tools/src/main/java/org/apache/kafka/tools/ProducerPerformance.java

```diff
@@ -350,7 +351,7 @@ public Stats(long numRecords, int reportingInterval) {
             this.start = System.currentTimeMillis();
             this.windowStart = System.currentTimeMillis();
             this.iteration = 0;
-            this.sampling = (int) (numRecords / Math.min(numRecords, 500000));
+            this.sampling = numRecords / Math.min(numRecords, 500000);
```

Comment:
We could also limit the value before casting to avoid overflow:
```
this.sampling = (int) Math.min(Integer.MAX_VALUE, numRecords / Math.min(numRecords, 500000));
```

But I think switching `long` as you've elected makes more sense.


I know it's unrelated but just 4 lines below, `this.totalLatency` is being initialized twice, could you remove that?